### PR TITLE
Upgraded Go version an Go dependencies versions

### PR DESCRIPTION
**Pull request by iuliana**
**What**
----
Upgraded Golang version used to build this project 1.20 -> 1.22.5. 
Upgraded the following dependencies:

1. `github.com/urfave/cli/v2`:                           v2.25.0 -> v2.27.2
2. `golang.org/x/crypto`:                                  v0.7.0 -> v0.25.0
3. `k8s.io/client-go`:                                         v0.26.2 -> v0.30.2
4. `github.com/cpuguy83/go-md2man/v2`:   v2.0.2 -> v2.0.4 
5. `github.com/xrash/smetrics`:  v0.0.0-20201216005158-039620a65673 ->  v0.0.0-20240312152122-5f08fbb34913
6. `golang.org/x/sys`:                                        v0.6.0 -> v0.22.0
7. `golang.org/x/term`:                                      v0.6.0  -> v0.22.0 

**Why**
----
To remove some vulnerabilities reported by Windows Defender and Virus Total.

### CODE REVIEW

**Review by zan-mateusz — APPROVED** [review]
LGTM

File: cli/go.mod [hunk 1]
```diff
@@ -16,19 +16,19 @@
 // under the License.
 module github.com/apache/brooklyn-client/cli
 
-go 1.20
+go 1.22.5
```
Comment (jcabrerizo) [inline cli/go.mod:19]:
```suggestion
go 1.22
```

This makes the pipeline to fail with
```
invalid go version '1.22.5': must match format 1.23
```

Only the minor version must be defined

**Review by jcabrerizo — CHANGES_REQUESTED** [review]

**Comment by ahgittin** [comment]
@iuliana probably the docker build image needs updated to use this version of go, otherwise the build will fail?

**Comment by iuliana** [comment]
> @iuliana probably the docker build image needs updated to use this version of go, otherwise the build will fail?

oh...ok, I'll get right on it

**Pull request merged**